modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -186,39 +194,56 @@ public ObjectValidationRule updateObjectValidationRule(
 
 	@Override
 	@Transactional(readOnly = true)
-	public void validate(BaseModel<?> baseModel, long objectDefinitionId)
+	public void validate(
+			BaseModel<?> baseModel, long objectDefinitionId,
+			JSONObject payloadJSONObject, long userId)
 		throws PortalException {
 
 		if (baseModel == null) {
 			return;
 		}
 
-		Map<String, Object> values = new HashMap<>();
+		List<ObjectValidationRule> objectValidationRules =
+			objectValidationRuleLocalService.getObjectValidationRules(
+				objectDefinitionId, true);
+
+		if ((objectValidationRules == null) ||
+			ListUtil.isEmpty(objectValidationRules)) {
+
+			return;
+		}
+
+		Map<String, Object> values = HashMapBuilder.<String, Object>putAll(
+			baseModel.getModelAttributes()
+		).build();
 
 		if (baseModel instanceof ObjectEntry) {
-			values = HashMapBuilder.<String, Object>putAll(
-				baseModel.getModelAttributes()
-			).putAll(
-				_objectEntryLocalService.getValues((ObjectEntry)baseModel)
-			).build();
+			values.putAll(
+				_objectEntryLocalService.getValues((ObjectEntry)baseModel));
 		}
 		else {
-			values = HashMapBuilder.<String, Object>putAll(
-				baseModel.getModelAttributes()
-			).putAll(
-				_objectEntryLocalService.
-					getExtensionDynamicObjectDefinitionTableValues(
-						_objectDefinitionPersistence.fetchByPrimaryKey(
-							objectDefinitionId),
-						GetterUtil.getLong(baseModel.getPrimaryKeyObj()))
-			).putAll(
-				EntityExtensionThreadLocal.getExtendedProperties()
-			).build();
+			Map<String, Serializable> extendedProperties =
+				EntityExtensionThreadLocal.getExtendedProperties();
+
+			if (extendedProperties != null) {
+				values.putAll(extendedProperties);
+			}
 		}
 
-		List<ObjectValidationRule> objectValidationRules =
-			objectValidationRuleLocalService.getObjectValidationRules(
-				objectDefinitionId, true);
+		ObjectDefinition objectDefinition =
+			_objectDefinitionPersistence.fetchByPrimaryKey(objectDefinitionId);
+
+		values.putAll(
+			_objectEntryLocalService.
```

Comment:
Can we use the payloadJSONObject.get("extendedProperties")?

Reply:
We can not use this to ObjectEntry because this properties is not created in payloadJSONObject.